### PR TITLE
Weekly Patch Release v1.6.1

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

```bash
gh pr list -s merged -S 'merged:2022-03-20T18:30:00.000Z..2022-04-12T18:30:00.000Z' --json mergedAt,milestone,url,mergeCommit,title --jq 'sort_by(.mergedAt) | reverse | .[] | select((.milestone.title == "1.6.x")) | [.mergedAt, .url, .mergeCommit.oid, .title] | join(" ")' --limit 100
```

```console
2022-04-12T21:28:54Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12608 77a02234e91a2646739074942198bcd33ffef662 Support auto_select_gpus with accelerator and devices api
2022-04-12T09:27:14Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12727 6fcb590cfb46291febd449bb336b917a06d9b2ab Update deepspeed precision test
2022-04-11T18:01:18Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12708 91c3d8ecb9a91c0ab197309f1c0bf12d53b1fb5d Raise `MisconfigurationException` when the accelerator is available but…
2022-04-11T15:29:54Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12677 f9b69ce5b0de32bfd2300eb8028fdbebf23fe5a1 CI: check docker requires
2022-04-11T15:10:33Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12676 3686778b89f584c3aacde0f4f5d89cc978e1f244 fix import failed
2022-04-11T13:38:49Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12563 f4883d6ead6e95ba767b64da1164ae1cd3cb3864 Run main progress bar independent of val progress bar in `TQDMProgressBar`
2022-04-11T13:38:27Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12707 cf0e3c62505405b21d17669c154aab755da2bf2b Fix a flaky test, that is not consistent on some configurations
2022-04-11T12:00:49Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12704 2f2a516de9e5bbda91be7916bf08c99d0ca223a8 Update LightningCLI tests to reflect changes in jsonargparse 4.6.0
2022-04-09T11:48:33Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12645 c233731b7c55d2533ab76eecf609e8df0e0ebfc3 Fix a typo in warning message inside Trainer.reset_train_dataloader
2022-04-09T10:41:37Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12672 2e5b3b333d5be5ba782c9c127d6ab21b3cb723e0 Allow self-reviewing to rerun gatekeeper
2022-04-08T19:40:19Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12669 a44b5dc0cb1489a805b73d0a786d34b9dea80c86 Don't raise a warning when `nn.Module`s are not saved under hparams
2022-04-08T09:25:51Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12661 f6d61cb08a9f9570cf8b2a11227bff4fa573fc31 Update ci_pr-gatekeeper.yml
2022-04-07T15:44:28Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12561 3bd48b85356e36b7a56b5d8f669403f0bfb9d717 Clear env before test cases requiring empty env
2022-04-07T15:02:26Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12646 7988351d8898f02e537a3c75f4e238985a018cba add PR Gatekeeper
2022-04-07T12:52:07Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12613 03609bc09cf231c79e3c14abf53d2066b85d1f1a [pre-commit.ci] pre-commit suggestions
2022-04-07T10:46:55Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12587 6aa8e26a4e2522e22b237213ad8d187db91e3793 Update `rank_zero_only` decorator for LSF environments
2022-04-07T07:32:10Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12629 800580a131bd109d9c1ae3fb3b519f021c0531f1 Fix `LightningLite.run` signature for arbitrary arguments
2022-04-06T16:47:36Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12609 0a72efd210d1c3d3a93f86cca096abee2130e5b1 Expose deprecated arguments from logger base interface
2022-04-06T15:01:25Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12627 58b6d19839bbc4de07416cb4dc87e2e437c02a6c Set a less strict pre-commit Python version
2022-04-06T14:42:32Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12618 2cf24a9e307ad809535cce6113dff681b09e611d Fix rich main progress bar update
2022-04-04T14:24:48Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12585 28877acef2376255e2c41c69ff86b2b8d5fa8249 Enable rank zero utilities tests that were never running
2022-04-04T13:08:14Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12449 e48f136088e3deb8d8d0ba81edc62232de7aed00 Run CircleCI jobs in seperate workflows to enable rerunning individually
2022-04-04T04:01:08Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12542 a2cdfc9e4f1e5e1a161210dc0a21842f3a76486b Bump version of black to 22.3.0
2022-04-02T05:34:25Z https://github.com/PyTorchLightning/pytorch-lightning/pull/12452 63f97e292548c31218617a3680b766b2aec15cd9 Avoid calling average_parameters multiple times per optimizer step
```

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

**Comment by kaushikb11** [comment]
Would like to have #12608 part of this release. It's priority!

**Comment by rohitgr7** [comment]
> 

yep. I'll monitor new PRs once again at EOD. If it shows up, will add it here else will be added in the next one.

**Review by Borda — APPROVED** [review]

**Comment by awaelchli** [comment]
#12608 should be ready now :)

**Review by awaelchli — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]
LGTM!

**Review by krshrimali — APPROVED** [review]

**Comment by rohitgr7** [comment]
> FYI, the failure in `Test with Conda` should be due to [actions/checkout#760](https://github.com/actions/checkout/issues/760).

see @Borda! I told you it might have to do something with gH actions 😄
